### DISCUSSION TITLE
Is it possible to convert a Prisma where object to where statement that can be used with $queryRaw?

### DISCUSSION THREAD

**Discussion by Hugo-Persson** [prisma/prisma#20292] [Q&A] [answered]
I have an `WhereInput` object like this:

```ts
function generateFilterForPrisma(params: FilterData): Prisma.SMTPLogWhereInput {
	const filterObj: Prisma.SMTPLogWhereInput = {};

    if (params.searchTerm) {
        const filter: Prisma.StringFilter = { contains: params.searchTerm };
        filterObj.OR = [
            { subject: filter },
			{ smtpHost: filter },
            { tag: filter },
			{ from: filter },
			{ recipient: {email: filter}},
			...(parseInt(params.searchTerm)? [{ recipient:  {userId: parseInt(params.searchTerm)} }] : []),
        ];
    }
	if (params.startDate) {
		const filter: Prisma.DateTimeFilter = { gte: new Date(params.startDate)};
		filterObj.createdAt = filter;
	}
	if (params.endDate) {
		const filter: Prisma.DateTimeFilter = { lte: new Date(params.endDate)};
		filterObj.createdAt = filter;
	}
	if (params.env) filterObj.env = params.env;
	
	if (params.userType ) filterObj.recipient = {userType: params.userType}; 
	
	return filterObj;
}

```

I would like to convert this to a `WHERE` clause that I can use with a `$queryRaw` statement. Does prisma have something built in for this?

**Comment by nurul3101** [comment]
Hello @Hugo-Persson 👋 

I believe you would be using this `WhereInput` in one of the Prisma queries. One solution I can think of is to enable logging with log level set to query ([Reference](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#log-to-stdout)). This should log the underlying SQL query that prisma generated, you can then extract the SQL Where clause from the logged query.

Hope this helps!

**Reply by Hugo-Persson** [reply]
Thank you for quick answer @nurul3101 ❤️ 

Is it possible to get the query in logging without executing the query? As I understand the page you sent it will log all queries when they are executed. I would like to cancel prisma from executing so I do not waste DB resources to get the WHERE clause

**Reply by nurul3101** [reply, marked as answer]
No, it's not possible to get the query without executing the query at the moment.
We have this related feature request that aims to achieve what you want to do:
- https://github.com/prisma/prisma/issues/5052

I would recommend you to add a 👍  to the request so that our product team can prioritise it.

Please let me know in case you have any other queries, if not, please consider marking the discussion as answered/closed. Thanks! 🙏

**Reply by Hugo-Persson** [reply]
Yes thank you 👍
